src/cmap/auth/auth_provider.ts

```diff
@@ -55,6 +55,6 @@ export class AuthProvider {
    */
   auth(context: AuthContext, callback: Callback): void {
     // TODO(NODE-3485): Replace this with MongoMethodOverrideError
-    callback(new MongoDriverError('`auth` method must be overridden by subclass'));
+    callback(new MongoAPIError('`auth` method must be overridden by subclass'));
```

Comment:
This would be runtime only because we don't expose auth provider to be overridden by users. Ideally this should be an abstract class and this method is abstract making TS force us to implement it. but for now we can just keep this error, but make it Runtime for accuracy.